lib/samson/secrets/vault_client_manager.rb

```diff
@@ -3,30 +3,33 @@
 module Samson
   module Secrets
     # Vault wrapper that sends requests to all matching vault servers
-    class VaultClient
+    class VaultClientManager
+      # TODO: make these configurable via env vars and send env vars to samson_secret_puller
+      MOUNT = 'secret'
+      PREFIX = 'apps'
+
       class VaultServerNotConfigured < StandardError
       end
 
-      def self.client
-        @client ||= new
+      def self.instance
+        @instance ||= new
       end
 
       # responsible servers should have the same data, so read from the first
       def read(id)
-        vault = responsible_clients(id).first
-        with_retries { vault.logical.read(wrap_id(id)) }
+        vault_client = responsible_clients(id).first
+        with_retries { vault_client.kv.read(id) }
       end
 
       # different servers have different ids so combine all
-      def list_recursive(path)
-        path = wrap_id(path)
-        all = Samson::Parallelizer.map(clients.values) do |vault|
+      def list_recursive(path = '')
```

Comment:
signature for list has a require path, so should have that here too for symmetry ?
https://github.com/hashicorp/vault-ruby/blob/cfc499b00e610804cbde9bebe64beb23296541d1/lib/vault/client.rb#L189

Reply:
Is that necessary for us though? In the context of Samson, we'll always be reading from something starting with `secret/apps`, so it made sense to me to have that as a default to save time / be more dry.

Reply:
nah, not necessary, was just thinking "if this wad part of the gem it should look like this"